packages/block-library/src/gallery/edit.js

```diff
@@ -142,6 +138,21 @@ class GalleryEdit extends Component {
 		} );
 	}
 
+	onAddMoreImages( newImages ) {
+		if ( ! window.event ) { // from "Upload" option after newImages absolute paths have been generated
```

Comment:
I'm not certain I understand why do we need this `if else` and I don't think having two separate functions is really needed here unless I'm missing something?

Reply:
@youknowriad thanks for the feedback.

Please see response to your questions below;

**1. why do we need this `if else`**
Without the `if else`, when one adds an image to the gallery using the MediaPlaceholder "Upload" button, 2 images are added instead of 1 causing duplicity.
We noticed this was because, during the file upload 2 steps happen. Step1: A temporary path is created. Step2: An absolute path is created after image is sucessfully uploaded. These 2 paths were both being picked and added to the gallery.
To stop the duplicity we used the `if else` to confirm that the upload event is complete and that's when we pick the new image's absolute path and add it to the gallery.

**2. I don't think having two separate functions is really needed here unless I'm missing something?**
We used two separate functions `onAddMoreImages()` and `updateImagesAttribute()` for code readability and ease debugging. Also in the future, one might want to call `updateImagesAttribute()` without `onAddMoreImages()`.